Сделал 2-ой пункт

lonelyfox-dev opened the pull request:
Реализована первая версия функции-обработчика входных параметров.

lonelyfox-dev committed:
Сделал 2-ой пункт

 -----

Реализована первая версия функции-обработчика входных параметров.

SHTYKOVYCH commented on lab1/parse-args.c:
```diff
@@ -0,0 +1,86 @@
+*/
+int parseArgs(int argc, char **argv, Args *args)
+{
+	args = (Args*) malloc(sizeof(Args));
```
В функции память выделять не надо. Я имел ввиду, что в основной функции создается структура, а в функцию передается указатель на нее.

SHTYKOVYCH commented on lab1/parse-args.c:
```diff
@@ -0,0 +1,86 @@
+
+	args.depack = 0;
+	args.outputFile = "archive";
+	args.inputFile = "";
```
Оно точно компилируется, args же указатель

SHTYKOVYCH commented on lab1/parse-args.h:
```diff
@@ -0,0 +1,3 @@
+#pragma once
```
По поводу этого надо будет уточнять, не факт, что такое позволят.  Лучше быть готовыми к переходу на C-style(ifndef и тд)

SHTYKOVYCH commented on lab1/parse-args.c:
```diff
@@ -0,0 +1,86 @@
+	Возвращает 1 если всё хорошо.
+	Возвращает 0 в случае ошибки.
+
+	Алярма: сейчас код ошибки 0 - единый для всех случаев.
```
Как сделаю первый пункт, будет enum c кодами ошибок

SHTYKOVYCH commented on lab1/parse-args.c:
```diff
@@ -0,0 +1,86 @@
+    /*
+    	TODO:
+
+    	1) Подумать, можно ли избавиться от стены if-ов
```
Можно завести массив с допустимыми флагами, при встрече флага сразу чекать следующий аргумент строки, если флаг, то выдавать ошибку, иначе выкидывать ошибку. Так получится пробегать по всем аргументам, в т ч повторяющимся в строке.

lonelyfox-dev replied:
Я думал сделать пробег по массивам argv до предпоследнего, чтобы при обращении к следующему элементу, как ты предложил, не происходил segmentation сюрприз с границами массива. Но понял что можно встретить что-то по типу `./archive -i in -o out -d`, где `-d` тупо не будет обработан. _Может и так сойдёт? :)_

SHTYKOVYCH replied:
А что мешает прямо внутри цикла увеличить счетчик на 1 и чекнуть, не стал ли он больше размера входных параметров? Так и код компактнее получится, и править чуть легче будет. Будет достаточно сделать проверку из разряда srtcmp(argv[i], flag) и все. Под это можно отдельную функцию бахнуть, тут же в файле, чтобы избегать дублирования. Тогда и строка, которую ты кинул выше, будет ок как работать. В идеале все преобразуется к switch

lonelyfox-dev replied:
Идеала не будет, Коля принял ислам

lonelyfox-dev committed:
Внёс минорные поправки по ПР ревью

-----

- Исправил прагму на классический дефайн в хэдере

- Исправил обращения к структуре Args внутри функции

SHTYKOVYCH commented on lab1/parse-args.c:
```diff
@@ -0,0 +1,86 @@
+	args.outputFile = "archive";
+	args.inputFile = "";
+
+	if ((argc > 6) || (argc < 3))
```
А если препод попросит добавить еще фич

lonelyfox-dev committed:
Внёс оставшиеся поправки к МР

-----

- Поменял логику работы основной функции, добавил вспомогательную для обработки имени файла

- Добавил коды ошибок (а кое где и хэндлер)

lonelyfox-dev committed:
Исправлена доё... Исправлено замечание по поводу расширяемости числа фич

lonelyfox-dev committed:
Добавил структуру Args в хэдер
